crates/engine/tree/src/tree/mod.rs

```diff
@@ -937,22 +949,27 @@ where
                                 })) {
                                     error!("Failed to send event: {err:?}");
                                 }
+                                Ok(())
                             }
                             BeaconEngineMessage::TransitionConfigurationExchanged => {
                                 // triggering this hook will record that we received a request from
                                 // the CL
                                 self.canonical_in_memory_state
                                     .on_transition_configuration_exchanged();
+                                Ok(())
                             }
                         }
                     }
                 }
             }
-            FromEngine::DownloadedBlocks(blocks) => {
-                if let Some(event) = self.on_downloaded(blocks) {
+            FromEngine::DownloadedBlocks(blocks) => match self.on_downloaded(blocks) {
```

Comment:
this can be
```rust
FromEngine::DownloadedBlocks(blocks) => {
    if let Some(event) = self.on_downloaded(blocks)? {
        self.on_tree_event(event);
    }
}
```